src/error.rs

```diff
@@ -75,6 +80,13 @@ impl fmt::Display for FormatterError {
             Self::Internal(message, _) | Self::Query(message, _) => {
                 write!(f, "{message}")
             }
+            Self::Formatting(err) => {
+                write!(
+                    f,
+                    "Idempotence found this error when running topiary on its own output:\n\t"
```

Comment:
You may want to match `Self::Formatting(Idempotence)` and `Self::Formatting(_)` separately, with messages like this:

"The formatter did not produce the same result when invoked twice (idempotence check). This is an error. It would be helpful if you logged this at https://github.com/tweag/topiary"

"The formatter errored when trying to format the code twice (idempotence check). This probably means that the formatter produced invalid code, and is an error. It would be helpful if you logged this at https://github.com/tweag/topiary"

Reply:
+1 to make explicit that this is a bug in Topiary itself that should be reported. The current message is an improvement but still leaves a doubt to the user that maybe it comes from the OCaml code. (For instance if they use Topiary but don't know what the idempotence check is.)

Also +1 to make the difference between various bugs in the idempotence check. That being said, isn't it already the case through the report of the found error? Like in this case I understand that it is a parse error and therefore that the output of the first pass was not parseable, which is different from parseable but different from the output of the second pass.

Reply:
You are right that `Formatting(Idempotence)` should have the message that currently `Idempotence` has, and the other `Formatting(...)` a message in a different spirit.
Agree about calling to log on GitHub. This was not done in the error messages, but only in the comments previously, so I reproduced it. Will change now.

Reply:
After further thought, I think that `Formatting(Idempotence)` does not make sense, so I put the `Idempotence` error outside of the `Formatting` wrapper.